Gems/AtomLyIntegration/CommonFeatures/Code/Source/PostProcess/EditorModeFeedback/EditorEditorModeFeedbackSystemComponent.cpp

```diff
@@ -0,0 +1,268 @@
+            const MeshFeatureProcessorInterface::MeshHandle& meshHandle,
+            Data::Instance<RPI::Material> maskMaterial)
+        {
+            const auto& shaderAsset = maskMaterial->GetAsset()->GetMaterialTypeAsset()->GetShaderAssetForObjectSrg();
+            const auto& objectSrgLayout = maskMaterial->GetAsset()->GetObjectSrgLayout();
+            const auto maskMeshObjectSrg = RPI::ShaderResourceGroup::Create(shaderAsset, objectSrgLayout->GetName());
+
+            // Set the object id so the correct MVP matrices can be selected in the shader
+            const auto objectId = featureProcessor->GetObjectId(meshHandle).GetIndex();
+            RHI::ShaderInputNameIndex objectIdIndex = "m_objectId";
+            maskMeshObjectSrg->SetConstant(objectIdIndex, objectId);
+
+            //! Set the id to write to the entity mask texture.
+            RHI::ShaderInputNameIndex maskIdIndex = "m_maskId";
+            maskMeshObjectSrg->SetConstant(maskIdIndex, 1);
+            
+            maskMeshObjectSrg->Compile();
+
+            return maskMeshObjectSrg;
+        }
+
+        //! Gets the view for the specified scene.
+        static const RPI::ViewPtr GetViewFromScene(const RPI::Scene* scene)
+        {
+            const auto viewportContextRequests = RPI::ViewportContextRequests::Get();
+            const auto viewportContext = viewportContextRequests->GetViewportContextByScene(scene);
+            const RPI::ViewPtr viewPtr = viewportContext->GetDefaultView();
+            return viewPtr;
+        }
+
+        //! Creates the material for the mask pass shader.
+        static Data::Instance<RPI::Material> CreateMaskMaterial()
+        {
+            const AZStd::string path = "shaders/postprocessing/editormodemask.azmaterial";
+            const auto materialAsset = GetAssetFromPath<RPI::MaterialAsset>(path, Data::AssetLoadBehavior::PreLoad, true);
+            const auto maskMaterial = RPI::Material::FindOrCreate(materialAsset);
+            return maskMaterial;
+        }
+
+        //! Get the world transform for the specified entity.
+        static AZ::Transform GetWorldTransformForEntity(EntityId entityId)
+        {
+            AZ::Transform worldTM;
+            AZ::TransformBus::EventResult(worldTM, entityId, &AZ::TransformBus::Events::GetWorldTM);
+            return worldTM;
+        }
+
+        EditorEditorModeFeedbackSystemComponent::MeshHandleDrawPackets::~MeshHandleDrawPackets() = default;
+
+        void EditorEditorModeFeedbackSystemComponent::Reflect(AZ::ReflectContext* context)
+        {
+            if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+            {
+                serialize->Class<EditorEditorModeFeedbackSystemComponent, AzToolsFramework::Components::EditorComponentBase>()
+                    ->Version(0)
+                    ;
+
+                if (AZ::EditContext* ec = serialize->GetEditContext())
```

Comment:
nit: prefer `editContext` (coding standards mention to avoid single letter abbreviations like this)